src/tests/ftest/util/dmg_utils.py

```diff
@@ -437,115 +437,45 @@ def pool_query(self, pool):
-            pool_map = {
-                "uuid": 0,
-                "ntarget": 1,
-                "disabled": 2,
-                "leader": 3,
-                "version": 4
-            }
-            # Mapping of the pool space entries to the match[2|3] indices
-            space_map = {
-                "total": 6,
-                "free": 7,
-                "free_min": 8,
-                "free_max": 9,
-                "free_mean": 10
-            }
-            # Mapping of the second indices mappings to the first match indices
-            map_values = {
-                0: pool_map,
-                1: {"target_count": 5},
-                2: space_map,
-                3: space_map,
-                4: {"status": 11, "status2": 12, "status3": 13}
-            }
-            for index_1, match_list in enumerate(match):
-                if index_1 not in map_values:
-                    continue
-                for key, index_2 in map_values[index_1].items():
-                    if index_1 == 2:
-                        if "scm" not in data:
-                            data["scm"] = {}
-                        data["scm"][key] = match_list[index_2]
-                    elif index_1 == 3:
-                        if "nvme" not in data:
-                            data["nvme"] = {}
-                        data["nvme"][key] = match_list[index_2]
-                    elif index_1 == 4:
-                        if "rebuild" not in data:
-                            data["rebuild"] = {}
-                        data["rebuild"][key] = match_list[index_2]
-                    else:
-                        data[key] = match_list[index_2]
-        return data
+        # Sample output
+        # {
+        #     "response": {
+        #         "status": 0,
+        #         "uuid": "EDAE0965-7A6E-48BD-A71C-A29F199C679F",
+        #         "total_targets": 8,
+        #         "active_targets": 8,
+        #         "total_nodes": 1,
+        #         "disabled_targets": 0,
+        #         "version": 1,
+        #         "leader": 0,
+        #         "rebuild": {
+        #             "status": 0,
+        #             "state": "idle",
+        #             "objects": 0,
+        #             "records": 0
+        #         },
+        #         "scm": {
+        #             "total": 16000000000,
+        #             "free": 15999992320,
+        #             "min": 1999999040,
+        #             "max": 1999999040,
+        #             "mean": 1999999040
+        #         },
+        #         "nvme": {
+        #             "total": 32000000000,
+        #             "free": 31999950848,
+        #             "min": 3999993856,
+        #             "max": 3999993856,
+        #             "mean": 3999993856
+        #         }
+        #     },
+        #     "error": null,
+        #     "status": 0
+        # }
+        return self._get_json_result(("pool", "query"), pool=pool)
```

Comment:
Can we provide a provision where we can use both json and non json output if somebody calls pool_query().
Supplying a simple flag like pool_query(pool, json=True/False) should enable or disable json or non json output.